src/task.js

```diff
@@ -1,140 +1,183 @@
+        task.inputId = parent.id;
+        task.editBox = document.getElementById(`input-${task.inputId}`);
+        task.deleteBtn = document.getElementById('delete-icon');
+
+        task.editBox.focus();
+        const { value } = task.editBox;
+        task.editBox.value = '';
+        task.editBox.value = value;
+
+        task.createKeyUpEventListenerForEdit();
+        task.createBlurEventListener();
+        task.createDeleteEventListener();
+      });
+    });
+  }
+
+  static createCheckEventlisteners = () => {
+    const checks = document.querySelectorAll('.complete-check');
+    checks.forEach((check, index) => {
+      check.addEventListener('change', (e) => {
+        e.preventDefault();
+        if (Task.completedIndexes.includes(index)) {
+          Task.completedIndexes = Task.completedIndexes.filter((i) => i !== index);
+          document.getElementById(`${index}`).querySelector('.list-text').style.textDecoration = 'none';
+        } else {
+          Task.completedIndexes.push(index);
+          document.getElementById(`${index}`).querySelector('.list-text').style.textDecoration = 'line-through';
+        }
+        localStorage.setItem('completedItems', JSON.stringify(Task.completedIndexes));
+      });
+    });
+  }
+
+  static createCompleteRemovalListener = () => {
+    const removeBtn = document.getElementById('remove-complete');
+    removeBtn.addEventListener('click', (e) => {
+      e.preventDefault();
+      Task.completedIndexes.forEach((i) => {
+        Task.tasks = Task.tasks.filter((t) => t.index !== i);
       });
+      Task.addedItemsUL.innerHTML = '';
+      Task.updateIdsAfterRemoval();
+      Task.populateTasks();
+      Task.updateLocalStorage();
+    });
+  }
+
+  static updateIdsAfterRemoval = () => {
+    let count = 1;
+
+    Task.tasks.forEach((t) => {
+      t.index = count;
+      count += 1;
     });
+    Task.completedIndexes = [];
+    localStorage.setItem('completedItems', JSON.stringify([]));
   }
 
-  updateLocalStorage() {
-    localStorage.setItem('tasksLocalStorage', JSON.stringify(this.tasks));
+  static updateLocalStorage = () => {
+    localStorage.setItem('tasksLocalStorage', JSON.stringify(Task.tasks));
   }
 
   display = () => {
     document.getElementById('list-ul').innerHTML += `
       
       <li id=${this.index}>
-        <input type="checkbox">
-          <span class="list-text">${this.description}</span> 
-          <span class="spacer"></span>
-          <span class="drag-icon"><i class="fa fa-ellipsis-v" aria-hidden="true"></i></span>
+        <input ${Task.completedIndexes.includes(this.index) ? 'checked' : ''} type="checkbox" id="check-${this.index}" class="complete-check">
+        <span style="text-decoration:${Task.completedIndexes.includes(this.index) ? 'line-through' : 'none'}" class="list-text">${this.description}</span> 
+        <span class="spacer"></span>
+        <span class="drag-icon"><i class="fa fa-ellipsis-v" aria-hidden="true"></i></span>
```

Comment:
- [ ] Kindly make the local storage completed status change upon checking the checkbox. This would be able to monitor the individual state of the task.
[OPTIONAL] You can consider adding to the booklist if the input value of is not empty. 

### What I have on my end
![image](https://user-images.githubusercontent.com/86200356/200834146-5fed0f1a-94d9-451a-a555-c2decfd408bf.png)